test/integration/tcp_proxy_integration_test.cc

```diff
@@ -0,0 +1,125 @@
+#include "test/integration/tcp_proxy_integration_test.h"
+
+#include "common/network/utility.h"
+#include "common/ssl/context_manager_impl.h"
+
+#include "test/integration/ssl_utility.h"
+#include "test/integration/utility.h"
+#include "test/mocks/runtime/mocks.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+
```

Comment:
FWIW I'm not a fan of this. It will make test much more verbose since all namespace references will need to be fully specified which is rarely the case when testing code within the same namespace. If there is a wish for this I would recommend opening up a separate discussion and not doing this in the current PR.